src/System.Runtime/tests/System/Reflection/IsCollectibleTests.cs

```diff
@@ -0,0 +1,183 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Runtime.Loader;
+using Xunit;
+
+namespace System.Reflection.Tests
+{
+    public class TestAssemblyLoadContext : AssemblyLoadContext
+    {
+        private readonly string _assemblyName;
+        private readonly string _assemblyPath;
+
+        public TestAssemblyLoadContext(string assemblyName, string assemblyPath)
+            : base(true)
+        {
+            _assemblyName = assemblyName;
+            _assemblyPath = assemblyPath;
+        }
+        protected override Assembly Load(AssemblyName assemblyName) =>
+            assemblyName.Name == _assemblyName ? LoadFromAssemblyPath(_assemblyPath) : null;
```

Comment:
It seems it would be cleaner to return null always (and use alc.LoadFromAssemblyPath in the tests)